src/Tasks/Microsoft.NET.Build.Tasks/PreprocessPackageDependenciesDesignTime.cs

```diff
@@ -200,9 +204,9 @@ private void PopulateAssemblies()
         /// This is not meant to be a general mechanism for injecting files that can't be handled
         /// through the normal NuGet package mechanisms.
         /// </summary>
-        private void PopulatePreResolvedNetStandardAssemblies()
+        private void PopulateExistingReferenceItems()
         {
-            var preResolvedAssemblyDependencies = new List<ITaskItem>();
+            var existingReferenceItemDependencies = new List<ITaskItem>();
             foreach (var reference in References)
             {
                 var packageName = reference.GetMetadata("NuGetPackageId");
```

Comment:
I suspect you're doing more work than necessary for all the references that came from NuGet.  I suspect you're not `continue;`ing until the last case.  Would it make sense to check `Type` metadata and `continue;` early if that was set?